File: backend/internal/models/user_admin.go
package models

// CreateUserRequest 创建用户请求
type CreateUserRequest struct {
	Username   string `json:"username" binding:"required"`
	Host       string `json:"host,omitempty"`        // MySQL only
	Password   string `json:"password" binding:"required"`
	IsSuperuser bool `json:"is_superuser,omitempty"` // PostgreSQL only
}

// ChangePasswordRequest 修改密码请求
type ChangePasswordRequest struct {
	Username    string `json:"username" binding:"required"`
	Host        string `json:"host,omitempty"` // MySQL only
	NewPassword string `json:"new_password" binding:"required"`
}

// DeleteUserRequest 删除用户请求
type DeleteUserRequest struct {
	Username string `json:"username" binding:"required"`
	Host     string `json:"host,omitempty"` // MySQL only
}

File: frontend/src/types/userAdmin.ts
/**
 * 用户管理相关类型定义
 */

export interface DatabaseUser {
  username: string
  host?: string // MySQL only
  role?: string // PostgreSQL only
  privileges?: string
  created_at?: string
}

export interface CreateUserRequest {
  username: string
  host?: string // MySQL only
  password: string
  is_superuser?: boolean // PostgreSQL only
}

export interface ChangePasswordRequest {
  username: string
  host?: string // MySQL only
  new_password: string
}

export interface DeleteUserRequest {
  username: string
  host?: string // MySQL only
}
